Kyrychenko/quizEntity/src/main/java/filters/QuizFilter.java

```diff
@@ -0,0 +1,103 @@
+package filters;
+
+import model.User;
+import service.UserService;
+
+import javax.servlet.*;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Objects;
+
+public class QuizFilter implements Filter {
+    private static final String PARAM_NAME_USERNAME = "name";
+    private static final String PARAM_NAME_LOGIN = "login";
+    private static final String PARAM_NAME_PASSWORD = "password";
+
+    private static final String ERROR_EMPTY_LOGIN_PASS_MESSAGE = "One of the entered values are empty";
+    private static final String ERROR_LOGIN_PASS_MESSAGE = "Incorrect login or password";
+    private static final String ERROR_LOGIN_EXIST_MESSAGE = "User with such login already exist";
+
+    private static final String LOGIN_PAGE = "/login.jsp";
+    private static final String REGISTRATION_PAGE = "/registration.jsp";
+
+    private static UserService userService = UserService.getInstance();
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain) throws IOException, ServletException {
+
+        HttpServletRequest request = (HttpServletRequest) servletRequest;
+        HttpServletResponse response = (HttpServletResponse) servletResponse;
+
+        HttpSession session = request.getSession();
+        String action = request.getParameter("action");
+
+        if (session.getAttribute("userID") != null) {
```

Comment:
It is not good practice, you move servlet logic to filter.
Filter just need to check, is user available to see this page, or page is forbidden.
And then pass control to servlet.
Please, refactor